src/metabase/lib/schema/ref.cljc

```diff
@@ -15,44 +16,54 @@
    [:map
     [:temporal-unit {:optional true} ::temporal-bucketing/unit]]])
 
+(mr/def ::base-type
+  [:fn #(isa? % :type/*)])
+
+(mr/def ::field.literal.options
+  [:and
+   ::field.options
+   [:map
+    [:base-type ::base-type]]])
+
 ;;; `:field` clause
+(mr/def ::field.literal
+  [:tuple
+   [:= :field]
+   ::field.literal.options
+   ::common/non-blank-string])
+
+(mr/def ::field.id
+  [:tuple
```

Comment:
Making these tuples definitely seems to fix some broken stuff, because with `:cat` or `:catn` because it is a regex schema if you don't wrap it in `:schema` it will splice it in to other "seqex" schemas